File: api/app/schemas/candidate.py
from typing import List
from pydantic import BaseModel, EmailStr
from typing import Optional

class CandidateCreate(BaseModel):
    name: str
    email: EmailStr
    phone: str = None

class CandidateResponse(BaseModel):
    id: str
    name: str
    email: EmailStr
    phone: str

    class Config:
        from_attributes = True



# Schema com dados opcionais para atualização
class CandidateUpdate(BaseModel):
    name: Optional[str] = None
    email: Optional[EmailStr] = None
    phone: Optional[str] = None



class CandidatePaginatedResponse(BaseModel):
    page: int
    limit: int
    count: int
    total_candidates: int
    candidates: List[CandidateResponse]

    class Config:
        from_attributes = True

File: web/src/types/candidate.ts
export interface Candidate {
    id: string;
    name: string;
    email: string;
    phone: string;
    created_at: string;
}


export interface CandidateResponse {
    id: string;
    name: string;
    email: string;
    phone: string;
}


export interface CandidatePaginatedResponse {

    page: number;
    limit: string;
    count: number;
    total_candidates: number;
    candidates: CandidateResponse[]

}
